martinLi/assignments/cyclicSort/lc268/lc268.js

```diff
@@ -0,0 +1,77 @@
+// Problem Statement #
+
+// We are given an array containing ‘n’ distinct numbers taken from the range 0 to ‘n’.
+//Since the array has only ‘n’ numbers out of the total ‘n+1’ numbers,
+//find the missing number.
+/*
+running through the list of numbers(loop) i to nums.length-1:
+num_swap = i
+check if the number we are on is > (length of the list - 1)
+    swap that number with the number that is at (length of the list - 2)// the very last number
+else if(nums[i] != num_swap){ //i = 0, nums[i] = 3 nums[3]
+  swap the number between nums[i] and nums[nums[i]] 
+}*/
+
+//[nums[i], (nums[nums[i]] = [nums[nums[i]], nums[i]])];
+
+const find_missing_number = function (nums) {
+  //for (let i = 0; i < nums.length; i++) {
+  let i = 0;
+  while (i < nums.length) {
+    let num_swap = nums[i];
+    /*if (nums[i] === nums.length) {
+      [nums[i], nums[nums.length - 1]] = [nums[nums.length - 1], nums[i]];
+    } else*/
+
+    if (nums[i] != nums.length && i !== nums[i]) {
+      [nums[i], nums[num_swap]] = [nums[num_swap], nums[i]];
+      //num_swap += 1;
+    } else {
+      i += 1;
+    }
+  }
+  for (i in nums) {
+    if (i != nums[i]) {
+      return i;
+    }
+  }
+  console.log(nums);
+
+  // TODO: Write your code here
+  return -1;
+};
```

Comment:
Opinionated topic: many will say to remove the inline comments from the finished program